src/nanoFramework.IoT.Device.CodeConverter/GenericsTemplates/Span.cs

```diff
@@ -0,0 +1,186 @@
+namespace System
+{
+    /// <summary>
+    /// Provides a type- and memory-safe representation of a contiguous region of arbitrary array.
+    /// </summary>
+    [Serializable, CLSCompliant(false)]
+    public readonly ref struct Span<T>
+    {
+        private readonly T[] _array;
+        private readonly int _start;
+        private readonly int _length;
+
+        /// <summary>
+        /// Creates a new Span object over the entirety of a specified array.
+        /// </summary>
+        /// <param name="array">The array from which to create the System.Span object.</param>
+        public Span(T[] array)
+        {
+            _array = array;
+            _length = array?.Length ?? 0;
+            _start = 0;
+        }
+
+        /// <summary>
+        /// Creates a new Span object that includes a specified number of elements
+        /// of an array starting at a specified index.
+        /// </summary>
+        /// <param name="array">The source array.</param>
+        /// <param name="start">The index of the first element to include in the new System.Span</param>
+        /// <param name="length">The number of elements to include in the new System.Span</param>
+        /// <exception cref="System.ArgumentOutOfRangeException">
+        /// array is null, but start or length is non-zero. -or- start is outside the bounds
+        /// of the array. -or- start and length exceeds the number of elements in the array.
+        /// </exception>
+        public Span(T[] array, int start, int length)
+        {
+            if (array != null)
+            {
+                if ((start + length > array.Length) || (start >= array.Length))
```

Comment:
should be this: `if ((length > array.Length - start) || (start > array.Length))`

Check things here: https://github.com/nanoframework/CoreLibrary/blob/develop/nanoFramework.CoreLibrary/System/SpanByte.cs

Reply:
Can start be equal to length, unless length=0?

Reply:
This looks correct:
```
if ((start + length > array.Length) || (start > array.Length) || ( start == array.Length && array.Length > 0))
```